batch7crud-roll10/src/main/java/com/lftechnology/batch7crud/dao/StudentDao.java

```diff
@@ -0,0 +1,175 @@
+            e.printStackTrace();
+            throw new DataException(e.getMessage());
+        } catch (NamingException e) {
+            e.printStackTrace();
+            throw new DataException(e.getMessage());
+        }
+
+    }
+
+
+    public void delete(Integer studentId) throws DataException {
+        try {
+            Connection conn = DbConnection.getConnection();
+
+            String query = "delete from student where id=?";
+            PreparedStatement ps = conn.prepareStatement(query);
+            ps.setInt(1,studentId);
+            ps.executeUpdate();
+
+        } catch (NamingException e) {
+            e.printStackTrace();
+            throw new DataException(e.getMessage());
+        } catch (SQLException e) {
+            e.printStackTrace();
+            throw new DataException(e.getMessage());
+        }
+    }
+
+
+    public void update(Student stud) throws DataException {
+        //update the student record
+        try {
+            Connection conn = DbConnection.getConnection();
+
+            String studentQuery = "UPDATE student set name=?, address=?, dob=?, department=?, batch=?, roll=? WHERE id=?";
+            PreparedStatement ps1 = conn.prepareStatement(studentQuery);
+            ps1.setString(1,stud.getName());
+            ps1.setString(2,stud.getAddress());
+            ps1.setDate(3, new java.sql.Date(stud.getDob().getTime()));
+            ps1.setString(4,stud.getDepartment());
+            ps1.setString(5,stud.getBatch());
+            ps1.setInt(6,stud.getRoll());
+            ps1.setInt(7,stud.getId());
+            ps1.executeUpdate();
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+            throw new DataException(e.getMessage());
+        } catch (NamingException e) {
+            e.printStackTrace();
+            throw new DataException(e.getMessage());
+        }
+    }
+
+
+    public Student fetchById(Integer id) throws DataException {
+        try {
+            String query = "select * from student where id=?";
+            Connection conn = DbConnection.getConnection();
+            PreparedStatement ps = conn.prepareStatement(query);
+            ps.setInt(1,id);
+
+            ResultSet studentResult = ps.executeQuery();
+
+            if (studentResult.next()){
+                Student student = new Student();
+                student.setId(studentResult.getInt("id"));
+                student.setRoll(studentResult.getInt("roll"));
+                student.setDepartment(studentResult.getString("department"));
+                student.setBatch(studentResult.getString("batch"));
+
+                student.setName(studentResult.getString("name"));
+                student.setAddress(studentResult.getString("address"));
+                student.setDob(studentResult.getDate("dob"));
+
+                return student;
+            }else{
+                return null;
+            }
+
+        } catch (NamingException e) {
+            e.printStackTrace();
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1148)